solutions/cf-702C.mdx

```diff
@@ -59,9 +59,86 @@ int main() {
 	cout << minR << endl;
 }
 ```
+			    
+</CPPSection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class CellularNetwork {
+	static final int TWO_BILLION = 2000000000;
+
+	// returns the first index in the array that is >= value,
+	// or towers.length if no such index exists
+	static int firstAtLeast(int[] towers, int value) {
+		int low = 0, high = towers.length;
+		while (low < high) {
+			int mid = (low + high) / 2;
+			if (towers[mid] >= value) {
+				high = mid;
+			}
+			else {
+				low = mid + 1;
+			}
+		}
+		return low;
+	}
+
+	public static void main (String[] args) {
+		Kattio io = new Kattio();
+		int n = io.nextInt();
+		int m = io.nextInt();
+
+		int[] cities = new int[n];
+		int[] towers = new int[m];
+
+		for (int i = 0; i < n; i++) {
+			cities[i] = io.nextInt();
+		}
+
+		for (int i = 0; i < m; i++) {
+			towers[i] = io.nextInt();
+		}
+
+		int minR = 0;
+		for (int i = 0; i < n; i++) {
+			int towerRight = firstAtLeast(towers, cities[i]);
+			int towerLeft = towerRight - 1;
+
+			int minRForThisCity = TWO_BILLION;
+			// make sure that an index actually exists,
+			// if not, it will return the length of the
+			// array which is equal to m
+			if (towerRight < m) {
+				minRForThisCity = Math.min(minRForThisCity, towers[towerRight] - cities[i]);
+			}
+			if (towerLeft >= 0) {
+				minRForThisCity = Math.min(minRForThisCity, cities[i] - towers[towerLeft]);
```

Comment:
i think these lines are too long

Reply:
but it barely exceeds on only a few lines so i think it should be fine

Reply:
the lines wrap on my screen but ok